src/cosmic_ray/interceptors/pragma_no_mutate.py

```diff
@@ -0,0 +1,47 @@
+"""An interceptor that uses metadata: no mutate to determine when specific mutations
+should be skipped.
+"""
+import re
+from functools import lru_cache
+import logging
+
+from cosmic_ray.work_item import WorkerOutcome, WorkResult
+
+log = logging.getLogger()
+
+
+def intercept(work_db):
+    """Look for WorkItems in `work_db` that should not be mutated due to spor metadata.
+
+    For each WorkItem, find anchors for the item's file/line/columns. If an
+    anchor exists with metadata containing `{mutate: False}` then the WorkItem
+    is marked as SKIPPED.
+    """
+
+    @lru_cache()
+    def file_contents(file_path):
+        "A simple cache of file contents."
+        with file_path.open(mode="rt") as handle:
+            return handle.readlines()
+
+    re_is_mutate = re.compile(r'.*#.*pragma:.*no mutate.*')
+
+    for item in work_db.work_items:
+        if not item.module_path:
+            continue
+
+        lines = file_contents(item.module_path)
+        try:
+            line_number = item.end_pos[0] - 1
```

Comment:
Don't we run the risk of `line_number` being negative? Won't that give the wrong results later on (i.e. on line 38)?

Reply:
Line numbers seem to be 1-based en column position 0-based.